drakrun/drakrun/util.py

```diff
@@ -1,9 +1,58 @@
 import base64
 import os
+import re
 import sys
 import subprocess
+from dataclasses import dataclass, field
+from typing import Dict
 
 from karton2 import Config
+from dataclasses_json import dataclass_json, config
+
+hexstring = config(
+    encoder=lambda v: hex(v),
+    decoder=lambda v: int(v, 16),
+)
+
+
+@dataclass_json
+@dataclass
+class VmiOffsets:
+    # Fields correspond to output defined in
+    # https://github.com/libvmi/libvmi/blob/master/examples/win-offsets.c
+
+    win_ntoskrnl: int = field(metadata=hexstring)
+    win_ntoskrnl_va: int = field(metadata=hexstring)
+
+    win_tasks: int = field(metadata=hexstring)
+    win_pdbase: int = field(metadata=hexstring)
+    win_pid: int = field(metadata=hexstring)
+    win_pname: int = field(metadata=hexstring)
+    win_kdvb: int = field(metadata=hexstring)
+    win_sysproc: int = field(metadata=hexstring)
+    win_kpcr: int = field(metadata=hexstring)
+    win_kdbg: int = field(metadata=hexstring)
+
+    kpgd: int = field(metadata=hexstring)
+
+    def from_tool_output(output: str) -> 'VmiOffsets':
+        """
+        Parse vmi-win-offsets tool output and return VmiOffsets.
+        If any of the fields is missing, throw TypeError
+        """
+        offsets = re.findall(r'^([a-z_]+):(0x[0-9a-f]+)$', output, re.MULTILINE)
+        vals = {k: int(v, 16) for k, v in offsets}
+        return VmiOffsets(**vals)
+
+
+@dataclass_json
+@dataclass
+class RuntimeInfo:
+    vmi_offsets: VmiOffsets
+    inject_pid: int
+
+    def load(file_obj) -> 'RuntimeInfo':
```

Comment:
any reason to pass file_obj here instead of raw string?

Reply:
I wanted to follow the API of `json` module :thinking:

Reply:
`IO[AnyStr]` type hint then?

Reply:
:trudne:

Reply:
fix'd